lkmpg.tex

```diff
@@ -1347,14 +1347,14 @@ \section{sysfs: Interacting with your module}
 What is the current value of \cpp|myvariable| ?
 
 \begin{codebash}
-cat /sys/kernel/mymodule/myvariable
+sudo cat /sys/kernel/mymodule/myvariable
 \end{codebash}
 
 Set the value of \cpp|myvariable| and check that it changed.
 
 \begin{codebash}
-echo "32" > /sys/kernel/mymodule/myvariable
-cat /sys/kernel/mymodule/myvariable
+sudo sh -c 'echo "32" > /sys/kernel/mymodule/myvariable'
```

Comment:
Use the combination of `sudo` and `tee` commands. See https://askubuntu.com/questions/408340/is-there-any-significance-to-using-tee

Reply:
Thank you for your suggestion. I have updated the command to use the combination of sudo and tee to perform the write operation. Here is the revised command:
```c
echo "32" | sudo tee /sys/kernel/mymodule/myvariable
```
This ensures that the data is correctly written to the file requiring elevated permissions.